qiskit/aqua/operators/list_ops/summed_op.py

```diff
@@ -131,6 +132,15 @@ def reduce(self) -> OperatorBase:
         else:
             return cast(OperatorBase, reduced_ops)
 
+    def to_circuit(self) -> QuantumCircuit:
+        """ Returns the quantum circuit, representing the summed operator.
+        Returns:
+            The circuit representation of the summed operator.
+        Raises:
+            ExtensionError: if operator is not unitary
+        """
+        return self.to_matrix_op().to_circuit()  # type: ignore
```

Comment:
See question above about the type ignore.

Reply:
While tests were added in the other operators i.e on self.to_circuit_op() that it would convert to a single circuit there is no test here. While the logic converts each oplist item to a matrix_op and then adds the the result, as far as I can see from MatrixOp add, can be a SummedOp still in the case the MatrixOp has parameterized coeffs. Perhaps a unit test for SummedOp and this case would show it working or show that this aspect needs addressing still, as I did not see any unit test that covered this.

Reply:
Thanks for the comment, I just added tests for SummedOp.to_circuit

Reply:
By parameterized I had meant using Parameter for coeff. Try this out in the test. It does not work and while it fails maybe it would be better to have some check here, like is done for others that used to_circuit

```
        m_op1_a = MatrixOp(m1, Parameter('alpha'))
        m_op2_b = MatrixOp(m2, Parameter('beta'))

        ops = m_op1_a + m_op2_b  # SummedOp 
        circuit = ops.to_circuit()
```

Reply:
Thanks for the example @woodsp-ibm 

I ran it in debugger and the error is raised already by calling ```to_circuit()``` on MatrixOp with parameter:
```python
from qiskit.aqua.operators import MatrixOp
from qiskit.circuit import Parameter
import numpy as np

m = np.array([[0, 0, 1, 0], [0, 0, 0, -1], [1, 0, 0, 0], [0, -1, 0, 0]])
mop = MatrixOp(m1, Parameter('alpha'))
mop.to_circuit()  # raises TypeError
```
```mop.to_instruction()``` or ```mop.to_matrix()``` will fail as well, and ```to_matrix()``` fails also for other PrimitiveOps with Parameter.

I just added the tests exposing these failures for PrimitiveOps. The error is raised, when ```complex``` or ```Operator``` is multiplied with ```Parameter```, while that operation is undefined.

Reply:
Did you try my specific example at all? That one tries, via the SummedOp to_circuit() to do it via the code `self.to_matrix_op().to_circuit()`. Now `self.to_matrix_op()` adds each operator together with the goal to have single matrix op that will end up as a circuit after conversion. But since the operators in the SummedOp are parameterized the add of them just returns another SummedOp. This then goes on recursively until it hits some recursive limit in Python and fails. Ideally, like you did the others where the `self.to_circuit_op().to_circuit() ` was split so the `self.to_circuit_op()` return type could be checked perhaps similar should be done for here too ie. check back types from `self.to_matrix_op()` so it doesn't end up in a recursive failure at least.

Reply:
Hi, no problem, its a lot to get into. Thanks for your patience and perseverance in doing this.

Reply:
My pleasure, thanks for all your help and support.